### PR TITLE
 Implement a new flag for running s_client in a non-interactive mode

### PR DESCRIPTION
**Pull request by ramikhaldi**
Hi,

This task aims to provide a new flag which can be used to run the s_client in a non-interactive mode. This can be helpful to test a  server without providing any user input. The process will be terminated after showing/verifying the server tls data.

Many developers want to test the server certificates or some TLS related stuff without any further interactions with the server. I have seen a lot of workarounds to achieve that, like piping QUIT, etc. But such approaches are not always trivial when switching the OS, executing the commands from different programming languages and so on..

I had to re-open a new pull request due to some branching issues I had.. Old conversations can be found here: https://github.com/openssl/openssl/pull/17096.

- [x] documentation is added or updated

Any feedback would be greatly appreciated. Thank you!

KR,
Rami

### CODE REVIEW

**Comment by beldmit** [comment]
You still have the same problems with this PR. Please remove x509.c at al from the PR.

**Comment by BenBE** [comment]
Can you condense these 9 commits down to the bare minimum (e.g. rebase/squash based on latest upstream master)? Looking through several "fix style" commits for changes that are overwritten later on just makes the reviewing more tedious than it needs to be.

**Comment by t8m** [comment]
And please drop the merge commit.

**Comment by ramikhaldi** [comment]
I got rid of the unrelated commits, let's see the ci results.

File: apps/s_client.c [hunk 1]
```diff
@@ -2942,6 +2948,13 @@ int s_client_main(int argc, char **argv)
                 goto shut;
             }
         }
+
+        /*don't wait for client input in the non-interactive mode*/
```
Comment (paulidale) [inline apps/s_client.c:2952]: spaces around the text:
`/* don't wait for client input in the non-interactive mode */`

**Review by paulidale — CHANGES_REQUESTED** [review]

**Comment by ramikhaldi** [comment]
I fixed a logical error. Can someone re trigger the CI, please. Thanks.

**Comment by beldmit** [comment]
BTW, why do we have have 'fips severity' label?

**Comment by beldmit** [comment]
I believe yes

File: apps/s_client.c (hunk 1 above)
Comment (ramikhaldi) [reply apps/s_client.c]: done.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you for your contribution.

**Pull request closed without merging**